contracts/ronin/BridgeTracking.sol

```diff
@@ -190,42 +206,59 @@ contract BridgeTracking is HasBridgeContract, HasValidatorContract, Initializabl
     address _bridgeOperator,
     bool _mustCountLastStats
   ) internal view returns (uint256 _totalBallots) {
-    _totalBallots = _periodStats[_period].totalBallotsOf[_bridgeOperator];
+    _totalBallots = _periodMetric[_period].totalBallotsOf[_bridgeOperator];
     if (_mustCountLastStats) {
-      _totalBallots += _temporaryStats.info.totalBallotsOf[_bridgeOperator];
+      _totalBallots += _bufferMetric.data.totalBallotsOf[_bridgeOperator];
     }
   }
 
   /**
-   * @dev Syncs period stats if the last epoch + 1 is already wrapped up.
+   * @dev Syncs period stats. Move all data from the buffer metric to a new period metric.
+   *
+   * Requirements:
+   * - The buffer epoch + 1 is already wrapped up.
    */
-  function _trySyncPeriodStats() internal {
+  function _trySyncBuffer() internal {
     uint256 _currentEpoch = _validatorContract.epochOf(block.number);
-    if (_temporaryStats.lastEpoch < _currentEpoch) {
-      (, uint256 _period) = _validatorContract.tryGetPeriodOfEpoch(_temporaryStats.lastEpoch + 1);
-      VoteStats storage _stats = _periodStats[_period];
-      _stats.totalVotes += _temporaryStats.info.totalVotes;
-      _stats.totalBallots += _temporaryStats.info.totalBallots;
-
-      address _voter;
-      for (uint _i = 0; _i < _temporaryStats.info.voters.length; _i++) {
-        _voter = _temporaryStats.info.voters[_i];
-        _stats.totalBallotsOf[_voter] += _temporaryStats.info.totalBallotsOf[_voter];
-        delete _temporaryStats.info.totalBallotsOf[_voter];
+    if (_bufferMetric.lastEpoch < _currentEpoch) {
+      (bool _filled, uint256 _currentPeriod) = _validatorContract.tryGetPeriodOfEpoch(_bufferMetric.lastEpoch + 1);
```

Comment:
Looks like this reverts this PR: https://github.com/axieinfinity/ronin-dpos-contracts/pull/170